core/txpool/celo_rates.go

```diff
@@ -0,0 +1,93 @@
+package txpool
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/contracts"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var (
+	unitRate = big.NewRat(1, 1)
+)
+
+func GetRatesForState(config *params.ChainConfig, state *state.StateDB) common.ExchangeRates {
+	stateCopy := state.Copy()
+	caller := &core.CeloBackend{
+		ChainConfig: config,
+		State:       stateCopy,
+	}
+
+	rates, err := caller.GetExchangeRates()
+	if err != nil {
+		log.Error("Error trying to get exchange rates.", "cause", err)
+	}
+	return rates
+}
+
+func ConvertCurrency(exchangeRates common.ExchangeRates, val1 *big.Int, currencyFrom *common.Address, currencyTo *common.Address) *big.Int {
```

Comment:
Should we move this to the `ConvertCurrencyToGold` function?

Reply:
What do you mean? move it to the same package/file ?

Reply:
Yes, basically put all the conversion functions in a file/module.

Reply:
I grouped exchanges and comparison into common/exchange